cloudevents/working-drafts/xml-format.md

```diff
@@ -0,0 +1,367 @@
+    ..
+    <myextension xsi:type="ce:string">my extension value</myextension>
+    <myboolean xsi:type="ce:boolean">false</myboolean>
+```
+
+## 3. Data
+
+The data portion of a CloudEvent follows a similar model to that employed by
+the [JSON Format specification][json-format]. A `data` element MUST be used to
+encapsulate the payload.
+
+An `xsi:type` is used to discrimate the payload type and MUST be present.
+
+The following data representations are supported:
+
+### 3.1 Binary Data
+
+Binary data MUST be carried in an element with an defined type of `ce:binary`.
+
+Example:
+
+``` xml
+<data xsi:type="ce:binary">.........</data>
+```
+
+### 3.2 Text Data
+
+Text MUST be carried in an element with an defined type of `ce:string`.
+
+Example:
+
+``` xml
+<data xsi:type="ce:string">This is text</data>
+```
+
+### 3.3 XML Data
+
+XML data MUST be carried in an element with a defined type of `xs:any` with
+a single child XML element (with any required namespace definitions).
+
+The child business data XML elements MUST NOT reside in the CloudEvent namespace,
+an empty namespace MAY be used as appropriate.
+
+Example:
+
+``` xml
+<data xsi:type="xs:any">
+    <myData xmlns="http://my.org/namespace">
+        ....
+    </myData>  
+</data>
+```
+
+## 4. Envelope
+
+Each CloudEvent is wholly represented as an XML element `<event>` that
+MUST carry the `specversion` as an XML attribute value.
+
+Such a representation MUST use the media type `application/cloudevents+xml`.
+
+The enveloping element contains:
+
+* A set of CloudEvent context attribute XML elements.
+* An OPTIONAL `data` XML element.
+
+Example _(XML preamble and namespace definitions omitted for brevity)_:
+
+``` xml
+<event specversion="1.0">
+    <time>2020-03-19T12:54:00-07:00</time>
+    <datacontenttype>text/plain</datacontenttype>
+    <id>000-1111-2222</id>
+    <source>urn:uuid:123e4567-e89b-12d3-a456-426614174000</source>
+    <type>SOME.EVENT.TYPE</type>
+    <myboolean xsi:type="ce:boolean">false</myboolean>
+    <data xsi:type="ce:string">Now is the winter of our discount tents...</data>
+</event>
+```
+
+## 5. XML Batch Format
+
+In the _XML Batch Format_ several CloudEvents are batched into a single XML
+element. The element comprises a list of elements in the XML Format.
+
+Although the _XML Batch Format_ builds on top of the _XML Format_, it is
+considered as a separate format: a valid implementation of the _XML Format_
+doesn't need to support it. The _XML Batch Format_ MUST NOT be used when only
+support for the _XML Format_ is indicated.
+
+An XML Batch of CloudEvents MUST use the media type
+`application/cloudevents-batch+xml`.
+
+Example _(XML preamble and namespace definitions omitted for brevity)_:
+
+```xml
+<batch>
+    <event specversion="1.0">
+       ....
+    </event>
+    <event specversion="1.0">
+       ....
+    </event>
+    ....
+</batch>
+```
+
+An example of an empty batch of CloudEvents (typically used in a response, but also valid in a request):
```

Comment:
I'd drop the comment about responses since events don't have responses and I don't think we want to get into that rat-hole.